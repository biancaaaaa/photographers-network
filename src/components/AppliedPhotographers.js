import React, {Component} from 'react';
import {Link} from 'react-router-dom';

export const AppliedPhotographers = ({photographers}) => (
  <div>
    <h3>Photographers who applied:</h3>
    {photographers.map((photographer, key) => {
      return <AppliedPhotographerElement key={key} uid={photographer.uid} name={photographer.displayName}/>
    })}
  </div>
);

const AppliedPhotographerElement = ({uid, name}) => (
  <div className="user-card applied">
    <h3>{name}</h3>
    <Link to={`/profile/${uid}`}>Visit Profile</Link>
    <div>
      <button className="gb-btn gb-btn-small gb-btn-primary">Accept</button>
      <button className="gb-btn gb-btn-small gb-btn-primary">Decline</button>
    </div>
  </div>
);

