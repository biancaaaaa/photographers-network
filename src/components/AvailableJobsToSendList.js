import React from "react";
import { PropTypes } from "prop-types";

export const AvailableJobsToSendList = ({
  existingJobs,
  sendRequestHandler,
  reqSentLoading,
  backHandler,
  photographerId,
  company
}) => (
  <React.Fragment>
    {!reqSentLoading && (
      <h2 className="send-job-offer-back" onClick={backHandler}>
        BACK
      </h2>
    )}
    {existingJobs.length == 0 ? (
      <h2>There are no jobs, at least no open jobs</h2>
    ) : reqSentLoading ? (
      <h2>Sending....</h2>
    ) : (
      <ul className="openjobslist">
        {existingJobs.map(el => (
          <li className="openjobitem" key={el.jobId}>
            <div className="job-infos">
              <h2 className="openjobtitle">{el.title}</h2>
              <p className="openjobtitle">{el.descrition}</p>
              <p className="openjoblocation">{el.location}</p>
              <p className="openjobprice">{el.price}Â£</p>
            </div>
            <div
              onClick={() =>
                sendRequestHandler(photographerId, company, el.jobId)
              }
              className="sendoffer"
            >
              Send
            </div>
          </li>
        ))}
      </ul>
    )}
  </React.Fragment>
);

AvailableJobsToSendList.propTypes = {
  existingJobs: PropTypes.array.isRequired,
  sendRequestHandler: PropTypes.func.isRequired,
  reqSentLoading: PropTypes.bool.isRequired,
  backHandler: PropTypes.func.isRequired
};
