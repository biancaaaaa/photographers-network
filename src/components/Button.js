import React from 'react';
import PropTypes from 'prop-types';

export const Button = ({clickHandler, classes, children}) => {
    return (
        <a onClick={clickHandler} className={classes}>{children}</a>
    );
};

Button.propTypes = {
    classes: PropTypes.string.isRequired
};