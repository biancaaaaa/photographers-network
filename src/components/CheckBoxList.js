import React from 'react';
import PropTypes from 'prop-types';

export const CheckBoxList = ({list , checkboxChangeHandler , filterName}) => (
 <React.Fragment>
  {list.map(el=>(
   <label key={el}>
    <input type='checkbox' data-value={el} data-for={`${filterName}Filter`} onChange={checkboxChangeHandler} />
    {el}
   </label>
  ))}
 </React.Fragment>
)

CheckBoxList.propTypes = {
 list : PropTypes.array,
 checkboxChangeHandler : PropTypes.func,
 filterName : PropTypes.string,
}