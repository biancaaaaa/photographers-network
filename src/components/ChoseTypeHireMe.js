import React from "react";
import { PropTypes } from "prop-types";

export const ChoseTypeHireMe = ({ typeHandler }) => (
  <div className="hire-me-type-selector">
    <div onClick={() => typeHandler("new-job")} className="hire-me-type-one">
      <h2>Create a new job offer.</h2>
      <div className="my-fency-button">Select</div>
    </div>
    <div
      onClick={() => typeHandler("existing-job")}
      className="hire-me-type-one"
    >
      <h2>Select from one of your jobs.</h2>
      <div className="my-fency-button">Select</div>
    </div>
  </div>
);

ChoseTypeHireMe.propTypes = {
  typeHandler: PropTypes.func.isRequired
};
