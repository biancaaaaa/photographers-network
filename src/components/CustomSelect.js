import React from "react";

export const CustomSelect = ({ optionsList , optionSelectHandler, showCustomSelect}) => (
  <ul className="custom-select-dropdown" style={{display: showCustomSelect?'block':'none'}}>
    {
			optionsList.map((el)=>(
				<li key={el} onClick={()=>optionSelectHandler(el)} className='custom-select-option'>{el}</li>
			))
		}
  </ul>
);
