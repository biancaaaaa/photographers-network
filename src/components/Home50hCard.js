import React from "react";
import { RoundExpandButton } from "./svg/RoundExpandButton";
import PropTypes from "prop-types";
import SignUp from "../contents/shared/sign-up/SignUp";

export default class Home50hCard extends React.Component {
  state = {
    expanded: false
  };

  expandHandler = () => {
    this.setState(prevState => ({
      expanded: !prevState.expanded
    }));
    // The way we set up a click event which will be registered in the google analythics
    // ReactGA.event({
    //   category: "User expanded div",
    //   action: "Click"
    // });
  };

  render() {
    const { backgroundUrl, titleContent, paragraphContent, type } = this.props;
    const { expanded } = this.state;
    return (
      <div
        className={`gb-home-50h-card-wrapper ${expanded ? "expanded" : ""}`}
        style={{ backgroundImage: `url(${backgroundUrl})` }}
      >
        <div className="full-black-overlay gb-background-black-opacity-80 gb-absolute-full-size" />
        <div className="gb-home-50h-card">
          <div className="content-box">
            <div className="text-wrapper gb-text-white">
              {titleContent && (
                <h2 className="content-box-heading">{titleContent}</h2>
              )}
              {paragraphContent && (
                <p className="content-box-paragraph">{paragraphContent}</p>
              )}
            </div>
            <RoundExpandButton
              clickHandler={this.expandHandler}
              classes="gb-icon-large gb-icon gb-icon-white"
            />
          </div>
        </div>
        {expanded && (
          <SignUp
            type={type}
            closeHandler={this.expandHandler}
            match={{ params: { type: "photographer" } }}
          />
        )}
      </div>
    );
  }
}

Home50hCard.propTypes = {
  backgroundUrl: PropTypes.string,
  titleContent: PropTypes.string,
  paragraphContent: PropTypes.string
};
