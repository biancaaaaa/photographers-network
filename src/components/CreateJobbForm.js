import React from 'react';
import { CustomSelect } from "./CustomSelect";
import { InputField } from './form/InputField';
import { NameInputSVG } from "./svg/NameInputSVG";
import { LocationSVG } from "./svg/LocationSVG";
import { MoneySVG } from './svg/MoneySVG';
import { CameraSVG } from "./svg/CameraSVG";
import { CalendarSVG } from "./svg/CalendarSVG";
import { TextArea } from './form/TextArea';
import PropTypes from 'prop-types';

const types = ["nature", "portrait", "dogs", "cats"];

export const CreateJobbForm = ({ 
 today,
 submitHandler, 
 showCustomSelectHandler, 
 changeHandler, 
 showCustomSelect, 
 optionSelectHandler, 
 jobbTitle, 
 jobbLocation, 
 jobbType, 
 jobbBudget,
 jobbDate, 
 jobbDescription}) => (
 <div className="create-jobb-page section-content with-padding">
  <h1>Create Job</h1>
  <form onSubmit={submitHandler}>
   <InputField
    svg={<NameInputSVG classes="gb-icon gb-icon-medium gb-icon-white inputIcon" />}
    value={jobbTitle}
    changeHandler={changeHandler}
    type='text'
    name='Title'
    placeholder='Name/Title'
   />
   <InputField
    svg={<LocationSVG classes="gb-icon gb-icon-medium gb-icon-white inputIcon" />}
    value={jobbLocation}
    changeHandler={changeHandler}
    type='text'
    name='Location'
    placeholder='Location'
   />
   <div
    className="custom-select gb-text-input gb-text-input-trans-background"
    onClick={showCustomSelectHandler}
   >
    <CameraSVG classes="gb-icon gb-icon-medium gb-icon-fill-white inputIcon" />
    {jobbType}
    <CustomSelect
     showCustomSelect={showCustomSelect}
     optionsList={types}
     optionSelectHandler={optionSelectHandler}
    />
   </div>
   <InputField
    svg={<MoneySVG
     classes="gb-icon gb-icon-medium gb-icon-fill-white inputIcon" />}
    value={jobbBudget}
    changeHandler={changeHandler}
    type='number' name='Budget'
    placeholder='Budget' />
   <InputField
    svg={<CalendarSVG
     classes="gb-icon gb-icon-medium gb-icon-fill-white inputIcon" />}
    value={jobbDate || today}
    changeHandler={changeHandler}
    type='date' name='Date' min={today} />
   <TextArea
    svg={<NameInputSVG
     classes="gb-icon gb-icon-medium gb-icon-white inputIcon" />}
    value={jobbDescription}
    name='Description'
    changeHandler={changeHandler}
    placeholder='Job description' />
   <input
    className='gb-btn gb-btn-large gb-btn-primary'
    type="submit"
    value="Create" />
  </form>
 </div>
);

CreateJobbForm.propTypes = {
  today : PropTypes.string.isRequired,
  submitHandler : PropTypes.func.isRequired, 
  showCustomSelectHandler : PropTypes.func.isRequired, 
  changeHandler : PropTypes.func.isRequired, 
  showCustomSelect : PropTypes.bool, 
  optionSelectHandler: PropTypes.func, 
  jobbTitle : PropTypes.string, 
  jobbLocation : PropTypes.string, 
  jobbType : PropTypes.string, 
  jobbBudget : PropTypes.string,
  jobbDate : PropTypes.string, 
  jobbDescription : PropTypes.string
};