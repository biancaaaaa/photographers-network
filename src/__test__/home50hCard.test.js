import React from "react";
import { shallow } from "enzyme";
import { Home50hCard } from "../components/Home50hCard";
import { RoundExpandButton } from "../components/svg/RoundExpandButton";

describe("Home50hCard", () => {
  const shallowRender = props => shallow(<Home50hCard {...props} />);

  it("always render a round expand button", () => {
    const el = shallowRender();
    expect(el.find(RoundExpandButton).length).toBe(1);
  });

  describe("No props passed", () => {
    const el = shallowRender();
    it("does not render a header", () => {
      expect(el.find("h2.content-box-heading").length).toBe(0);
    });
    it("does not render a paragraph", () => {
      expect(el.find(".content-box-paragraph").length).toBe(0);
    });
  });

  describe("titleContent passed", () => {
    const headerProp = "My amazing Title";
    const el = shallowRender({ titleContent: headerProp });

    it("renders a header", () => {
      expect(el.find("h2.content-box-heading").length).toBe(1);
    });
    it("header has the speciffied value, from props", () => {
      expect(el.find("h2.content-box-heading").text()).toBe(headerProp);
    });
  });

  describe("paragraphContent passed", () => {
    const paragraphProp = "Hello there";
    const el = shallowRender({ paragraphContent: paragraphProp });

    it("renders the paragraph", () => {
      expect(el.find(".content-box-paragraph").length).toBe(1);
    });
    it("paragraph has the text value, from the props", () => {
      expect(el.find(".content-box-paragraph").text()).toBe(paragraphProp);
    });
  });
});
