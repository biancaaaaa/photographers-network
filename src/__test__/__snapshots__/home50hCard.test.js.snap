// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home50hCard matches snapshot 1`] = `
<div
  className="gb-home-50h-card-wrapper "
  style={
    Object {
      "backgroundImage": "url(e.jpg)",
    }
  }
>
  <div
    className="full-black-overlay gb-background-black-opacity-80 gb-absolute-full-size"
  />
  <div
    className="gb-home-50h-card"
  >
    <div
      className="content-box"
    >
      <div
        className="text-wrapper gb-text-white"
      >
        <h2
          className="content-box-heading"
        >
          test title
        </h2>
        <p
          className="content-box-paragraph"
        >
          test paragraph
        </p>
      </div>
      <svg
        className="gb-icon-large gb-icon gb-icon-white"
        height="100%"
        id="round-expand-button"
        onClick={[Function]}
        viewBox="0 0 72 72"
        width="100%"
      >
        <title>
          expand button
        </title>
        <g>
          <path
            d="M35.863 69.226C54.2889 69.226 69.226 54.2889 69.226 35.863C69.226 17.4371 54.2889 2.5 35.863 2.5C17.4371 2.5 2.5 17.4371 2.5 35.863C2.5 54.2889 17.4371 69.226 35.863 69.226Z"
            stroke="white"
            strokeWidth="5"
          />
          <path
            d="M37.2021 28.384H34.4541V44.464H37.2021V28.384Z"
            fill="white"
          />
          <path
            d="M43.868 37.692V34.944H27.788V37.692H43.868Z"
            fill="white"
          />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
